### ISSUE TITLE
missing-parentheses-for-call-in-test not raised as expected

### ISSUE THREAD

**Issue by carlio** [pylint-dev/pylint#8680]
### Bug description

**Preamble**

I recently had a bug in my code which was not caught by pylint (or any other linter or my IDE). I had a method on a class which I had intended to mark as a `@property` but had forgotten, and then later used `if instance.that_property`

**Reproduction steps**

I have added a test to the `missing-parentheses-for-call-in-test` functional tests to show a case where I would have expected pylint to raise "missing-parentheses-for-call-in-test" but does not.

(I will create a draft PR after this and link to it)

### Command used

```shell
On my project, I simply ran `pylint`.
```

### Pylint output

```shell
It outputs other linting warnings, but not `missing-parentheses-for-call-in-test` as expected
```

### Expected behavior

`missing-parentheses-for-call-in-test` should be raised for cases such as the example I showed in my reproduction steps.

### Pylint version

```shell
>> pylint --version
pylint 2.17.4
astroid 2.15.4
Python 3.11.0 (main, Oct 25 2022, 13:54:35) [GCC 9.4.0]

# However this happens with the current `main` branch too.
```

**Comment by BrianHVB** [comment]
I just encountered the same error. This came up in our production code and I found/enabled the missing-parentheses-for-call-in-test (W0126) rule.

Here is some example code that should cause the rule to issue a warning but currently does not:

```
class Thing:
    def __init__(self):
        self.status = 'off'

    def is_on(self):
        return self.status == 'on'


if __name__ == '__main__':
    my_thing = Thing()
    if my_thing.is_on:
        print('it is on')
```
